services/filesstore/s3store.go

```diff
@@ -238,10 +238,15 @@ func (b *S3FileBackend) ListDirectory(path string) (*[]string, *model.AppError)
 	}
 
 	doneCh := make(chan struct{})
-
 	defer close(doneCh)
 
-	for object := range s3Clnt.ListObjects(b.bucket, path, false, doneCh) {
+	updatedPath := path
+	if !strings.HasSuffix(path, "/") && len(path) > 0 {
```

Comment:
Any unit tests for s3 we can update here?

Reply:
yep already updated ;-)
https://github.com/mattermost/mattermost-server/pull/11654/files#diff-f86aca3f016fa314f6ce7b800efdb74aR252